game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/change/RemoveUnprotectedUnitsTest.java

```diff
@@ -0,0 +1,510 @@
+package games.strategy.triplea.delegate.battle.steps.change;
+
+import static games.strategy.triplea.Constants.UNIT_ATTACHMENT_NAME;
+import static games.strategy.triplea.delegate.battle.FakeBattleState.givenBattleStateBuilder;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenAnyUnit;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitIsSea;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitTransport;
+import static games.strategy.triplea.delegate.battle.steps.MockGameData.givenGameData;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyCollection;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.math.BigDecimal;
+import java.util.List;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class RemoveUnprotectedUnitsTest {
+
+  @Mock ExecutionStack executionStack;
+  @Mock IDelegateBridge delegateBridge;
+  @Mock BattleActions battleActions;
+  @Mock GamePlayer attacker;
+  @Mock GamePlayer defender;
+  @Mock Territory battleSite;
+
+  @Nested
+  class GetNames {
+
+    @Test
```

Comment:
A @DisplayName on this one could be useful IMO to explain why there is no name expected if land. It would be good to see this made clear up front why we expect this situation to be true. I mention that because without context, it's not clear why 'unprotected units' is an invalid step for land, 'land' can have protected or unprotected units, hence there is context here into the production code needed to understand the test, which is a test code maintenance scaling problem.

Reply:
This makes me wonder if the rule should be named more directly "RemoveUnprotectedCasualtyRestrictedTransports", WDYT?

Reply:
Do you mean rename the step?  Or the test?

And I agree that the `getNames` condition on water vs land is weird but that is the relic of the original code.  I'm not confident in changing at this point.  I actually think that this step should be split in 2 steps, one for the undefended transports and one for the unprotected units.  But with the discussion we had in the forum about the "auto-kill" logic, I think this step will need to be reworked completely.  So, I'd rather leave the original logic alone and test it.

Reply:
I was more suggesting to rename the  class under test.